static/src/javascripts/projects/common/modules/weather.js

```diff
@@ -124,17 +127,32 @@ define([
                 });
         },
 
+        fetchForecastData: function (location) {
+            return self.getWeatherData(config.page.forecastsapiurl + '/' + location.id + '.json')
+                .then(function (response) {
+                    self.renderForecast(response);
+                }).fail(function (err, msg) {
+                    raven.captureException(new Error('Error retrieving forecast data (' + msg + ')'), {
+                        tags: {
+                            feature: 'weather'
+                        }
+                    });
+                });
+        },
+
         bindEvents: function () {
-            bean.on($('.js-weather-input')[0], 'click', function () {
+            bean.on($('.js-weather-input')[0], 'click', function (e) {
+                e.preventDefault();
                 self.toggleControls(true);
             });
-            bean.on($('.js-close-location')[0], 'click', function () {
+            bean.on($('.js-close-location')[0], 'click', function (e) {
+                e.preventDefault();
                 self.toggleControls(false);
             });
-            bean.on($('.js-weather-input')[0], 'blur', function () {
-                self.toggleControls(false);
+            bean.on($('.js-toggle-forecast')[0], 'click', function (e) {
+                self.toggleForecast(e);
             });
-            mediator.on('autocomplete:fetch', this.fetchData);
+            mediator.on('autocomplete:fetch', this.fetchWeatherData);
```

Comment:
It's 2015! Why not use `this.fetchWeatherData.bind(this)` and then not have to use all of this messy `self.func()` closure patterns all over this module.